src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java

```diff
@@ -1079,24 +1079,29 @@ private void terminateParticipant(
 
         synchronized (participantLock)
         {
-            if (participant.isSessionEstablished())
+            try
             {
-                JingleSession jingleSession = participant.getJingleSession();
+                if (participant.isSessionEstablished())
+                {
+                    JingleSession jingleSession = participant.getJingleSession();
 
-                jingle.terminateSession(jingleSession, reason, message, sendSessionTerminate);
+                    jingle.terminateSession(jingleSession, reason, message, sendSessionTerminate);
 
-                removeSources(
-                    jingleSession,
-                    participant.getSourcesCopy(),
-                    participant.getSourceGroupsCopy(),
-                    false /* no JVB update - will expire */,
-                    sendSourceRemove);
+                    removeSources(
+                        jingleSession,
+                        participant.getSourcesCopy(),
+                        participant.getSourceGroupsCopy(),
+                        false /* no JVB update - will expire */,
+                        sendSourceRemove);
 
-                participant.setJingleSession(null);
+                    participant.setJingleSession(null);
+                }
+            }
+            finally
```

Comment:
Is this the only thing that needs to be handled in the `finally`?  What happens to the participant's bridge session in this throw case?

Reply:
another solution is to add "isconnected" checks to anything that goes to the network, so that the cleanup can happen gracefully

Reply:
> What happens to the participant's bridge session in this throw case?

@JonathanLennox 👍 thanks, I've adjusted the code to suppress the errors and move on to terminate the bridge session. It'll eventually fail, but at least it will remove the participant from the `bridgeSession.participants` list and try to remove the bridge session.

> another solution is to add "isconnected" checks to anything that goes to the network, so that the cleanup can happen gracefully

@paweldomas 👋 being connected now doesn't imply it'll be connected when the underlying code runs, hence there's always a chance an exception will be thrown

Reply:
I meant this `at org.jitsi.impl.protocol.xmpp.colibri.ColibriConferenceImpl.updateChannelsInfo(ColibriConferenceImpl.java:821)` or this `at org.jitsi.jicofo.JitsiMeetConferenceImpl.removeSources(JitsiMeetConferenceImpl.java:1776)`

Reply:
or this `at org.jitsi.jicofo.JitsiMeetConferenceImpl$BridgeSession.updateColibriOctoChannels(JitsiMeetConferenceImpl.java:2731)`

anyway not a big deal